### PR TITLE
Solution v2

### PR DESCRIPTION
**Pull request by ealitten**

### CODE REVIEW

File: lib/water.rb [hunk 1]
```diff
@@ -1,5 +1,25 @@
 class Water
-  def find_water(data)
-    
+  def find_water(towers)
+    raise TypeError unless towers.kind_of?(Array)
```
Comment (korki) [inline lib/water.rb]: 👍

File: lib/water.rb [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+    water_collected = 0
+    towers.each_with_index { |tower_height,index|
```
Comment (korki) [inline lib/water.rb]: For multiline blocks use do..end
Space after coma

**Review by korki — COMMENTED** [review]
Nice and clean solution. It's missing a bit of general ruby coding style, but at least you're consistent ;)

**Comment by ealitten** [comment]
Oops, this was supposed to just fix the numbering in the spec, I forgot it would update with later commits!

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]
@ealitten good job! 

Next question: How does this solution in with your idea of OOP in ruby?

What would you change and why?

File: lib/water.rb (hunk 2 above)
Comment (dearshrewdwit) [reply lib/water.rb]: @korki ❤️

File: lib/water.rb (hunk 1 above)
Comment (dearshrewdwit) [reply lib/water.rb]: ermm 👎

Comment (dearshrewdwit) [reply lib/water.rb]: I forget the input type-check is actually in the spec 😱

**Pull request closed without merging**